### ISSUE TITLE
Deleting thumbnail in admin does not call fileService.delete

### ISSUE THREAD

**Issue by ripvannwinkler** [medusajs/medusa#4092]
### Describe the bug

When clicking the Delete icon on a product thumbnail in the admin UI, the product is updated but the thumbnail is not deleted via the file service.

### System information

Medusa version (including plugins): v1.10.1
Node.js version: v16.20.0
Database: postgres
Operating system: Windows 11
Browser (if relevant):

### Steps to reproduce the behavior

1. Add a thumbnail to a product.
2. Click the trash icon to delete the thumbnail.
3. Check the admin UI to see that the thumbnail does not exist for the product.
4. Check the file service backend to see that the thumbnail image still exists.

### Expected behavior

Updating the thumbnail to null should trigger a call to fileService.delete with the file key for the image.

**Comment by tsvetann** [comment]
I can confirm this big on a new setup V1.10. Same behavior when you try to replace the existing thumbnail with a new one. The old thumbnail isn't deleted

**Comment by ripvannwinkler** [comment]
Also seems to be an issue with regular media, not just thumbnails

**Comment by olivermrbl** [comment]
That's right. We currently don't delete files when you remove them from a product.

~~You can create a subscriber, that does the job for you:~~

**Comment by tsvetann** [comment]
@olivermrbl I noticed that you've recently updated the label for this issue from "bug" to "feature". If I understand correctly, it seems to imply that the functionality in question is not currently implemented and the plan is to add it as a new feature, correct?

As someone who is relatively new to this project and affected by this functionality, I'd be interested in trying my hand at implementing this feature. It seems like a great opportunity for me to get more familiar with the library so please let me know if this matches your plans.

**Comment by chemicalkosek** [comment]
@olivermrbl This will just give info about which fields changed, but we have no way of knowing which file to delete from the service. Or is there a way?

Honestly this feels more like a bug than a feature. We already have hundreds of orphaned images on remote service.

**Comment by tsvetann** [comment]
> @olivermrbl This will just give info about which fields changed, but we have no way of knowing which file to delete from the service. Or is there a way?
> 
> Honestly this feels more like a bug than a feature. We already have hundreds of orphaned images on remote service.

You are describing the case in which we replace existing image, right? So there seems to be two distinct cases: 
Thumbnails:
1. replace existing image. (hits /admin/uploads)
2. removing existing image (hits admin/products/:product with payload {thumbnail: null})

Media:
Deleting image posts to admin/products/:product with payload array of fully qualified url strings

**Comment by ripvannwinkler** [comment]
> Honestly this feels more like a bug than a feature. We already have hundreds of orphaned images on remote service.

Echoing this sentiment. I honestly thought it was an oversight. When media is deleted, it should be deleted. I realize that changing that now could impact current customers, but it should maybe at least be introduced as a config option.

At the very least, it's a hole in the docs. I'm not just sure where it would go.

**Comment by olivermrbl** [comment]
> This will just give info about which fields changed, but we have no way of knowing which file to delete from the service. Or is there a way?

You are right. I take my suggestion back. We are only publishing the keys of the updated fields :)

**Comment by ripvannwinkler** [comment]
On a related note, the `DeleteFileType` type referenced in the `delete` method of `AbstractFileService` shows `fileKey`, but the API seems to pass `file_key`. Not sure if I'm missing something or if the type declaration should be updated.

**Comment by olivermrbl** [comment]
> I noticed that you've recently updated the label for this issue from "bug" to "feature". If I understand correctly, it seems to imply that the functionality in question is not currently implemented and the plan is to add it as a new feature, correct?

Yes and no. It means the functionality is currently not implemented, but adding it would provide value to users. It does not mean that it's planned work. Many other features and improvements are taking higher priority for our core team. 

> As someone who is relatively new to this project and affected by this functionality, I'd be interested in trying my hand at implementing this feature

PRs are always welcome and highly appreciated, so please do! Though, this feature might need a bit of discussion, as there are multiple ways of going about it and things to be wary of. Now that you mention you want to familiarize yourself with our project, I suggest you dig a bit into the code, and then we can bounce ideas off each other afterward. How does that sound?

**Comment by tsvetann** [comment]
@olivermrbl sounds like a plan

**Comment by tsvetann** [comment]
@olivermrbl lets chat about how this can be implemented. What's your preferred way of communication?

**Comment by bacarybruno** [comment]
Hello there 👋 is there anything new about this topic ?
I currently have the same issue and started implementing a subscriber that will remove the files when the product is updated / deleted but fell like it's something that should be done on the core, and not with a custom subscriber.

So far I managed to delete files when the product is deleted, but the product updated event only send changed fields (btw the event also sends unchanged fields), so there is currently no way to remove the old files when product images are replaced for example. It could be done by sending old and new values of the product, but to be discussed.

I would also like to be involved or at least follow in case you start discussing about this feature @tsvetann / @olivermrbl 🙏

**Comment by tsvetann** [comment]
There has been no progress/interest from the core team on this topic so far

**Comment by bacarybruno** [comment]
I tried implementing the feature in this PR https://github.com/medusajs/medusa/pull/5129 currently in draft. I would like to have your opinion @tsvetann / @olivermrbl if you have time, to check if there are some considerations I have to take into account 🙏
